sdl_android/src/main/java/com/smartdevicelink/api/datatypes/SdlFile.java

```diff
@@ -16,6 +16,29 @@
     private FileType    fileType;
     private boolean     persistentFile;
 
+    public SdlFile(){}
+
+    public SdlFile(@NonNull String fileName, @NonNull FileType fileType, int id, boolean persistentFile){
+        setName(fileName);
+        setType(fileType);
+        setResourceId(id);
+        setPersistent(persistentFile);
+    }
+
+    public SdlFile(@NonNull String fileName, @NonNull FileType fileType, Uri uri, boolean persistentFile){
+        setName(fileName);
+        setType(fileType);
+        setUri(uri);
+        setPersistent(persistentFile);
+    }
+
+    public SdlFile(@NonNull String fileName, @NonNull FileType fileType, byte[] data, boolean persistentFile){
+        setName(fileName);
+        setType(fileType);
+        setFileData(data);
+        setPersistent(persistentFile);
```

Comment:
Since there is no synchronization in the set methods it doesn't make sense to call them here instead of setting the value directly. The method call is an unnecessary step.